ports/linux/datetime-init.c

```diff
@@ -15,9 +15,54 @@
 #include <stdint.h>
 #include <stdlib.h>
 #include <time.h>
+#include "bacnet/basic/service/h_ts.h"
 #include "bacport.h"
 #include "bacnet/datetime.h"
 
+
+static int32_t timedifference(struct timeval t0, struct timeval t1)
+{
+    return (t0.tv_sec - t1.tv_sec)*1000 + (t0.tv_usec - t1.tv_usec) / 1000;
+}
+
+/**
+ * @brief Set offset from the system clock.
+ * @param bdate BACnet Date structure to hold local time
+ * @param btime BACnet Time structure to hold local time
+ * @param utc - True for UTC sync, False for Local time
+ * @return True if time is set
+ */
+void datetime_timesync(
```

Comment:
please add this public API to src/bacnet/datetime.h file at the bottom with the other "implementation agnostic functions - create your own!"